cpp/ql/src/semmle/code/cpp/models/interfaces/FunctionInputsAndOutputs.qll

```diff
@@ -13,95 +13,227 @@ class ParameterIndex extends int {
   ParameterIndex() { exists(Parameter p | this = p.getIndex()) }
 }
 
-newtype TFunctionInput =
+private newtype TFunctionInput =
   TInParameter(ParameterIndex i) or
-  TInParameterPointer(ParameterIndex i) or
-  TInQualifier()
+  TInParameterDeref(ParameterIndex i) or
+  TInQualifierObject() or
+  TInQualifierAddress()
 
+/**
+ * An input to a function. This can be:
+ * - The value of one of the function's parameters
+ * - The value pointed to by one of function's pointer or reference parameters
+ * - The value of the function's `this` pointer
+ * - The value pointed to by the function's `this` pointer
+ */
 class FunctionInput extends TFunctionInput {
   abstract string toString();
 
-  predicate isInParameter(ParameterIndex index) { none() }
+  predicate isParameter(ParameterIndex index) { none() }
```

Comment:
I'm afraid we can't change these names without keeping the old names as `deprecated` synonyms. They're used by at least one customer: https://semmle.slack.com/archives/C7NS4H8HY/p1561548291277300

Reply:
It also means we need to write a change note about this.

Reply:
Done and done.